src/mscorlib/src/System/String.cs

```diff
@@ -1695,29 +1695,104 @@ private String CtorCharCount(char c, int count)
         private static unsafe int wcslen(char *ptr)
         {
             char *end = ptr;
+            
+            // TODO: Test to see if switching to something like
+            // the 64-bit version would be beneficial on x86 as well
+
+            // First make sure our pointer is aligned on a word boundary
+            int alignment = IntPtr.Size - 1;
+
+            // If ptr is at an odd address, this loop will simply iterate all the way
```

Comment:
How does that contradict what I wrote? Let's say this is on 64-bit, the address passed in ends in 0x6, and the string is meant chars long.. This loop will iterate through 1 char,  at  point it's  at an aligned  address, and then exit, not at the end.